src/js/pages/GranthIndex/components/GranthIndex.tsx

```diff
@@ -1,38 +1,150 @@
-import React, { useEffect, MouseEvent } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import { Link } from 'react-router-dom';
 import { Location } from 'history';
 
 import BreadCrumb from '@/components/Breadcrumb';
 import Fetch from '@/components/Fetch';
-import { raagIndices } from '../constants/raag-indices';
+import { raagIndices, IRaagIndex } from '../constants/raag-indices';
 import { toAngURL } from '../../../util';
 import { pageView } from '../../../util/analytics';
 import { AmritKeertanIndex } from './AmritKeertanIndex';
-import { sanitizeHash, scrollToHash } from '../util';
+import { sanitizeHash, scrollToHash, desanitzeHash } from '../util';
 import { TEXTS } from '../../../constants';
 
 interface IGranthIndexProps {
-  location: Location
+  location: Location;
 }
 
+const GRANTH_NAMES = raagIndices.map(({ name }) => name);
+
 export const GranthIndex: React.FC<IGranthIndexProps> = ({ location }) => {
   const { hash } = location;
 
+  const [selection, setSelection] = useState<string[]>(() =>
+    hash ? desanitzeHash(hash) : [GRANTH_NAMES[0]]
+  );
+
+  // Helps check if a list item is selected to show subSets and description
+  const isSelected = useCallback(
+    (name: string) => {
+      if (selection.includes(name)) {
+        return true;
+      }
+    },
+    [selection]
+  );
+
+  // Called onClick to show description and subSets
+  const changeSelection = useCallback(
+    (path: string[]) => {
+      if (path.join() == selection.join()) {
+        path.pop();
+      }
+      setSelection(path);
```

Comment:
@yudhv veerji, as much as I understood, we doing this pop() here to toggle the state of that list item. If it's open, and we click on that, we are moving it using pop? if so, that would work on last level list item as well ?

Reply:
@gurjit03 that is correct. We only pop when the current selection and onClicked selection are the same, to collapse the clicked list-item. Just checked, it works if a description is added to the vaars (i.e. it opens and closes as expected when clicked) which are the last item in the heirarchy.